From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gleb Pesin <dormancygrace@gmail.com>
Date: Tue, 1 Sep 2026 00:00:00 +0300
Subject: [PATCH] usb: ch9344: remove unused packet count

The write path calculates packnum but never reads it. Remove the dead
calculation so external-module builds are clean with W=1.

Signed-off-by: Gleb Pesin <dormancygrace@gmail.com>
---
 driver/ch9344.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/driver/ch9344.c b/driver/ch9344.c
index ee3ed46..8ceb052 100644
--- a/driver/ch9344.c
+++ b/driver/ch9344.c
@@ -1337,7 +1337,7 @@ static int ch9344_tty_write(struct tty_struct *tty,
 	int portnum = ch9344_get_portnum(tty->index);
 	int timeout;
 	int maxep = ch9344->writesize / 20;
-	int packnum, maxpacknum;
+	int maxpacknum;
 	int packlen, total_len, sendlen;
 
 	if (!count)
@@ -1354,11 +1354,6 @@ static int ch9344_tty_write(struct tty_struct *tty,
 	total_len = count;
 	sendlen = 0;
 
-	if (count % (maxep - 3))
-		packnum = count / (maxep - 3) + 1;
-	else
-		packnum = count / (maxep - 3);
-
 transmit:
 	spin_lock_irqsave(&ch9344->write_lock, flags);
 	wbn = ch9344_wb_alloc(ch9344, portnum);
-- 
2.43.0
